### PR TITLE
Fix asJson method to pass mapper in WebClientRequestPreparation

### LINKED ISSUE #5454: WebClientRequestPreparation.asJson not reusing ObjectMapper when specified
https://github.com/line/armeria/blob/main/core/src/main/java/com/linecorp/armeria/client/WebClientRequestPreparation.java#L243

The `WebClientRequestPreparation.asJson(Class<? extends T> clazz, ObjectMapper mapper)` call isn't passing the `mapper` on to the `ResponseAs.json` call as it should, so it ends up _not_ using the specified mapper class and instead constructs a new one.

### PR DESCRIPTION
**Pull request by YangSiJun528**
Motivation:

Resolve issue #5454 with proper handling of ObjectMapper instance.

Modifications:

- Pass the provided ObjectMapper to the ResponseAs.json method:
    `WebClientRequestPreparation.asJson(Class<? extends T> clazz, ObjectMapper mapper)`.

Result:

- The `asJson` method pass the provided ObjectMapper instance to the `ResponseAs.json` method as it should
- Closes #5454

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/client/WebClientRequestPreparation.java [hunk 1]
```diff
@@ -240,7 +240,7 @@ public <T> FutureTransformingRequestPreparation<ResponseEntity<T>> asJson(Class<
                                                                               ObjectMapper mapper) {
         requireNonNull(clazz, "clazz");
         requireNonNull(mapper, "mapper");
-        return asEntity(ResponseAs.json(clazz));
+        return asEntity(ResponseAs.json(clazz, mapper));
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/client/WebClientRequestPreparation.java:243]: Could you also add a unit test to detect unexpected changes in the future?

**Comment by minwoox** [comment]
@YangSiJun528 Gentle ping. 😉

**Comment by YangSiJun528** [comment]
Sorry for the delayed response. I will add the unit test as requested.

**Comment by YangSiJun528** [comment]
I mistakenly merged main into this PR branch, so I had to force reset it to undo the [changes](https://github.com/line/armeria/compare/be2bcd157ea3d6e8af08d958c630bba7ad9c1418..ae7754c74e69a25911c8f2a12c65e85d945e3542).

File: core/src/main/java/com/linecorp/armeria/client/WebClientRequestPreparation.java (hunk 1 above)
Comment (YangSiJun528) [reply core/src/main/java/com/linecorp/armeria/client/WebClientRequestPreparation.java:243]: I have added the test code in WebClientRequestPreparationTest.
You can review the changes in the following commit: 8bdcb3e.

However, I am not sure it helps detect unexpected changes in the future.
Could you let me know which test cases should be added?

File: core/src/test/java/com/linecorp/armeria/client/WebClientRequestPreparationTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            sb.service("/json", (ctx, req) -> HttpResponse.ofJson(new MyMessage("hello")));
+        }
+    };
+
+    private static WebClient client;
+
+    @BeforeAll
+    static void beforeAll() {
+        client = WebClient.of(server.httpUri());
+    }
```
Comment (minwoox) [inline core/src/test/java/com/linecorp/armeria/client/WebClientRequestPreparationTest.java:51]: I think we don't have to have this as a field. We could just use the client in the method.

File: core/src/test/java/com/linecorp/armeria/client/WebClientRequestPreparationTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        final ObjectMapper mapper = new ObjectMapper();
+        final CompletableFuture<ResponseEntity<MyMessage>> future2 = client.prepare()
+                .get("/json")
+                .asJson(MyMessage.class, mapper)
+                .execute();
+
+        final ResponseEntity<MyMessage> response2 = future2.join();
+
+        assertThat(response2.content()).isEqualTo(new MyMessage("hello"));
```
Comment (minwoox) [inline core/src/test/java/com/linecorp/armeria/client/WebClientRequestPreparationTest.java:73]: There's no different between response1 and response2.
I think we can use a different mapper for the second case to see the difference. For example:
```
final ObjectMapper mapper = new ObjectMapper();
mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
then, we can make the first response fail and the second response succeed.

**Comment by YangSiJun528** [comment]
I have updated the test code to reflect the review comments.
View changes: c7c23ea

**Review by minwoox — APPROVED** [review]
Looks great, thank @YangSiJun528! 😄

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

**Comment by minwoox** [comment]
Well done, @YangSiJun528! 😄

**Pull request merged**